### PR TITLE
Fixed an empty ByteStrings problem in InputStreamSinkStage

### PR DESCRIPTION
**Pull request by tkawachi**
InputStreamSinkStage returns an additional byte (0x01) when an empty ByteString comes from upstream. It breaks contents. Please see a test in this PR.

I modified to drop empty ByteStrings to fix an issue because we have no interest in empty ByteStrings.

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST

File: akka-stream/src/main/scala/akka/stream/impl/io/InputStreamSinkStage.scala [hunk 1]
```diff
@@ -25,7 +25,9 @@ private[stream] object InputStreamSinkStage {
   case object Close extends AdapterToStageMessage
 
   sealed trait StreamToAdapterMessage
-  case class Data(data: ByteString) extends StreamToAdapterMessage
+  case class Data(data: ByteString) extends StreamToAdapterMessage {
+    require(data.nonEmpty)
+  }
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/impl/io/InputStreamSinkStage.scala]: not sure we need this check -- it has runtime cost after all, and we always avoid creating an empty one in any case now it seems anyway already on line 80?

Comment (tkawachi) [reply akka-stream/src/main/scala/akka/stream/impl/io/InputStreamSinkStage.scala]: It's just for readability. It might be enough to comment.

Comment (ktoso) [reply akka-stream/src/main/scala/akka/stream/impl/io/InputStreamSinkStage.scala]: Yes, please change to that :)

File: akka-stream/src/main/scala/akka/stream/impl/io/InputStreamSinkStage.scala [hunk 2]
```diff
@@ -127,7 +131,7 @@ private[stream] object InputStreamSinkStage {
 
   @scala.throws(classOf[IOException])
   override def read(): Int = {
-    val a = Array[Byte](1)
+    val a = new Array[Byte](1)
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/impl/io/InputStreamSinkStage.scala]: :+1:

File: akka-stream/src/main/scala/akka/stream/impl/io/InputStreamSinkStage.scala [hunk 3]
```diff
@@ -127,7 +131,7 @@ private[stream] object InputStreamSinkStage {
...
     if (read(a, 0, 1) != -1) a(0) & 0xff
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/impl/io/InputStreamSinkStage.scala]: would it be better to be more explicit?

```
read(a, 0, 1) match {
  case 1 => a(0) // & 0xff ?
  case 0 => 0 // is this unexpected, should throw?
  case -1 => -1
  case len => throw new IllegalStateException(s"Invalid length [$len]")
}
```

**Review by patriknw — APPROVED** [review]
LGTM

**Review by johanandren — APPROVED** [review]
LGTM

**Comment by johanandren** [comment]
Fixes #24880

**Pull request merged**